rabbitmq-native/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java

```diff
@@ -131,144 +139,88 @@ private static int getTimeValuesInMillis(BDecimal value) {
     }
 
-    private static SSLContext getSSLContext(BMap<BString, Object> secureSocket) {
-        try {
-            // keystore
-            KeyManagerFactory keyManagerFactory = null;
-            if (secureSocket.containsKey(RabbitMQConstants.CONNECTION_KEYSTORE)) {
-                @SuppressWarnings("unchecked")
-                BMap<BString, Object> cryptoKeyStore =
-                        (BMap<BString, Object>) secureSocket.getMapValue(RabbitMQConstants.CONNECTION_KEYSTORE);
-                char[] keyPassphrase = cryptoKeyStore.getStringValue(RabbitMQConstants.KEY_STORE_PASS).getValue()
-                        .toCharArray();
-                String keyFilePath = cryptoKeyStore.getStringValue(RabbitMQConstants.KEY_STORE_PATH).getValue();
-                KeyStore keyStore = KeyStore.getInstance(RabbitMQConstants.KEY_STORE_TYPE);
-                if (keyFilePath != null) {
-                    try (FileInputStream keyFileInputStream = new FileInputStream(keyFilePath)) {
-                        keyStore.load(keyFileInputStream, keyPassphrase);
-                    }
-                } else {
-                    RabbitMQMetricsUtil.reportError(RabbitMQObservabilityConstants.ERROR_TYPE_CONNECTION);
-                    throw RabbitMQUtils.returnErrorValue(RabbitMQConstants.CREATE_SECURE_CONNECTION_ERROR +
-                                                                 "Path for the keystore is not found.");
-                }
-                keyManagerFactory =
-                        KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
-                keyManagerFactory.init(keyStore, keyPassphrase);
-            }
+    /**
+     * Creates and retrieves the SSLContext from socket configuration.
+     *
+     * @param secureSocket secureSocket record.
+     * @return Initialized SSLContext.
+     */
+    private static SSLContext getSSLContext(BMap<BString, Object> secureSocket)
+            throws IOException, CertificateException, KeyStoreException,
+            UnrecoverableKeyException, KeyManagementException, NoSuchAlgorithmException {
+        // Keystore
+        String keyFilePath = null;
+        char[] keyPassphrase = null;
+        char[] trustPassphrase;
+        String trustFilePath;
+        if (secureSocket.containsKey(RabbitMQConstants.CONNECTION_KEYSTORE)) {
+            @SuppressWarnings("unchecked")
+            BMap<BString, Object> cryptoKeyStore =
+                    (BMap<BString, Object>) secureSocket.getMapValue(RabbitMQConstants.CONNECTION_KEYSTORE);
+            keyPassphrase = cryptoKeyStore.getStringValue(RabbitMQConstants.KEY_STORE_PASS).getValue().toCharArray();
+            keyFilePath = cryptoKeyStore.getStringValue(RabbitMQConstants.KEY_STORE_PATH).getValue();
+        }
 
-            // truststore
+        // Truststore
+        @SuppressWarnings("unchecked")
+        BMap<BString, Object> cryptoTrustStore =
```

Comment:
Just curious, don't we need to check whether the truststore is provided or not? Or is it done before coming to this point?

Reply:
Truststore is a mandatory field in the `SecureSocket` record. Should we check if it's given? 🤔